docs/assets/js/charts.js

```diff
@@ -142,6 +142,110 @@ function createSpinner(canvas)
     };
 }
 
+function aggregateTimeData(data, aggregationConfig)
+{
+    if (!(data instanceof Array))
+        throw 'expected data array as input';
+
+    if (data.length < 1)
+        return;
+
+    // Turn date strings into proper date objects
+    for (let i = 0; i < data.length; i++)
+        data[i]['date'] = d3.isoParse(data[i]['date']);
+
+    // Sort data, just in case it isn’t already
+    data.sort((row1, row2) => row1['date'] - row2['date']);
+
+    const dateStart = data[0]['date'];
+    // Ranges are exclusive, so add one more day to include the last date
+    const dateEnd = d3.utcDay.offset(data[data.length - 1]['date'], 1);
+
+    let period;
+
+    switch (aggregationConfig['period'])
+    {
+        case 'week':
+            period = d3.utcMonday;
+            break;
+        case 'month':
+            period = d3.utcMonth;
+            break;
+        default:
+            throw 'unknown aggregation period "' + aggregationConfig['period'] + '"';
+    }
+
+    // Don't use incomplete periods at the beginning and the end of the data
+    const t0 = period.ceil(dateStart);
+    // In d3, ranges include the start value but exclude the end value
+    // We want to include the last period as well, so add one more period
+    const t1 = period.offset(period.floor(dateEnd), 1);
+    const periods = period.range(t0, t1);
+
+    let aggregatedData = Array();
+
+    for (let i = 0; i < periods.length - 1; i++)
+    {
+        const t0 = periods[i];
+        const t1 = periods[i + 1];
+
+        // Note that this assumes complete data in the period
+        // Should data points be missing, aggregation methods such as the sum will lead to results that can't be
+        // compared to periods with complete data
+        // Hence, the maintainers of the data need to ensure that the input is well-formed
```

Comment:
micro-nit: It might be more readable if you add a `.` at the end of the sentence.

Reply:
I didn’t add periods, because we don’t have them in comments in other places. Should we do this everywhere then? Or put periods just *between* sentences in multisentence comments?

Reply:
Okay, then I’d say that we put periods *between* sentences systematically *except* for the end of comments (as is practiced by many print magazines, for example).

Reply:
I changed this as I suggested above.